### PR TITLE
added fees collector original caller functionality

### PR DESCRIPTION
**Pull request by psorinionut**

### CODE REVIEW

File: energy-integration/fees-collector/src/lib.rs [hunk 1]
```diff
@@ -64,8 +70,8 @@ pub trait FeesCollector:
                 let locked_rewards = self.lock_virtual(
                     self.get_base_token_id(),
                     reward.amount,
-                    caller.clone(),
-                    caller.clone(),
+                    orig_caller.clone(),
```
Comment (dorin-iancu) [inline energy-integration/fees-collector/src/lib.rs:73]: Must send tokens to caller.
```suggestion
                    caller.clone(),
```

File: energy-integration/fees-collector/src/lib.rs [hunk 2]
```diff
@@ -78,7 +84,7 @@ pub trait FeesCollector:
         }
 
         if !output_rewards.is_empty() {
-            self.send().direct_multi(&caller, &output_rewards);
+            self.send().direct_multi(&orig_caller, &output_rewards);
```
Comment (dorin-iancu) [inline energy-integration/fees-collector/src/lib.rs:87]:
```suggestion
            self.send().direct_multi(&caller, &output_rewards);
```

Comment (psorinionut) [reply energy-integration/fees-collector/src/lib.rs]: Done.

File: energy-integration/fees-collector/src/lib.rs (hunk 1 above)
Comment (psorinionut) [reply energy-integration/fees-collector/src/lib.rs]: Done.

**Review by dorin-iancu — DISMISSED** [review]

**Pull request closed without merging**